### PR TITLE
Added Elasticsearch::get_documents and Indexable::multi_get

### PR DESCRIPTION
**Pull request by felipeelia**
### Description of the Change

This is a "take over" of #1959

> We needed an easy way to fetch multiple documents per API call, so we added get_documents to the Elasticsearch class and then added multi_get to the Indexable class for ease of use.
> 
> Props @nickdaugherty
> cc: @rinatkhaziev @pschoffer @netsuso @parkcityj

### Alternate Designs

> N/A

### Benefits

> Allows you to get multiple documents per API call.

### Possible Drawbacks

> N/A

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Changelog Entry

Added Elasticsearch::get_documents and Indexable::multi_get. Props @nickdaugherty and @felipeelia

### CODE REVIEW

File: includes/classes/Elasticsearch.php [hunk 1]
```diff
@@ -650,6 +650,56 @@ public function delete_network_alias( $alias ) {
...
+
+		$docs = [];
+
+		if ( is_array( $response['docs'] ) ) {
+			foreach ( $response['docs'] as $doc ) {
+				if ( ! empty( $doc['exists'] ) || ! empty( $doc['found'] ) ) {
+					$docs[] = $doc['_source'];
+				} else {
+					$docs[] = null;
+				}
```
Comment (Rahmon) [inline includes/classes/Elasticsearch.php:696]: @felipeelia What do you think about using the ID as an index in the $docs[]?

Now if we don't find a doc, the returned array looks like `[ 0 => doc1, 1 => null, 2 => doc3 ]`. Using the ID as an index the array would like `[ 11 => doc1, 25 => doc 3 ]` and we can check if a doc was found checking the index value.

Comment (felipeelia) [reply includes/classes/Elasticsearch.php]: makes sense @Rahmon . I've also added a new filter there so people can change it before it is returned.

**Review by Rahmon — APPROVED** [review]
LGTM :+1:

**Comment by felipeelia** [comment]
@oscarssanchez as all tests have passed, can you please give this a final review and merge it? Thanks!

**Review by oscarssanchez — APPROVED** [review]

**Pull request merged**